src/skydb.ts

```diff
@@ -70,9 +70,15 @@ export async function setJSON(
     try {
       const publicKey = pki.ed25519.publicKeyFromPrivateKey({ privateKey: privateKeyBuffer });
       entry = await this.registry.getEntry(toHexString(publicKey), dataKey, opts);
-      revision = entry.entry.revision + 1;
+      revision = entry.entry.revision + BigInt(1);
     } catch (err) {
-      revision = 0;
+      revision = BigInt(0);
+    }
+  } else {
+    // Assert the input is 64 bits.
+    const newint = BigInt.asUintN(64, revision);
+    if (newint != revision) {
+      throw new Error("Received revision number > 2^64-1");
```

Comment:
This is potentially dev facing error - we should denote that the 2^64-1 is the hard limit of 64 bit int.